plugins/widget/plugin.js

```diff
@@ -2177,6 +2177,16 @@
 							element.attributes[ 'data-cke-widget-data' ] = encodeURIComponent( JSON.stringify( data ) );
 							element.attributes[ 'data-cke-widget-upcasted' ] = 1;
 
+							// Preserve initial and trailing space by replacing white space with &nbsp; (#605).
```

Comment:
I am not sure if this method is the right place for such transformations. As it should be placed in `toHtml` listener as it is now, I think it should be put a little further (e.g. https://github.com/ckeditor/ckeditor-dev/pull/680/files#diff-d932030cd4e22ec8b0e30a362269b7e8R2806). The `createUpcastIterator` has different responsibility so it does not fit here.